openmdao/core/component.py

```diff
@@ -1548,7 +1548,14 @@ def _resolve_src_inds(self, my_tdict, top):
                                                 meta['src_indices'], src_shape)
                 elif _is_slicer_op(src_inds):
                     meta['src_slice'] = src_inds
-                    src_inds = _slice_indices(src_inds, np.prod(parent_src_shape), parent_src_shape)
+                    try:
+                        src_inds = _slice_indices(src_inds, np.prod(parent_src_shape),
+                                                  parent_src_shape)
+                    except IndexError as err:
+                        raise IndexError(f"In component {self.msginfo}: \nError '{err}'\n  in "
```

Comment:
No need for the "In component" part of this message.   All of our other error messages just start with `{self.msg_info}:`, which gives the class and name of the system/component..   This message would be inconsistent with that common format.